### ISSUE TITLE
Move the emulator into CC:Tweaked

### ISSUE THREAD

**Issue by SquidDev** [SquidDev-CC/copy-cat#34]
The problem with using copy-cat for tweaked.cc is that often the emulator doesn't support the latest documented features. For instance, we don't support https://github.com/cc-tweaked/CC-Tweaked/commit/4b0988768df2fd55623c94e3e4e177723b9c61d4, despite that being two weeks old now!

The alternative would be to move the TeaVM patches to their "actual" project:
 - For Cobalt, we could publish a separate jar with a "teavm" classifier.
 - For CC:T, move the Java and _core_ TS code into the the `projects/web` folder. tweaked.cc should then consume the output of that, rather than relying on copy-cat's embedding functionality.
 - copy-cat then just clones CC:T as a sub-module and does some build magic.

This however, does have some possible issues/caveats:
 - This does complicate CC:T dev, as anyone working on the core project now needs to worry about keeping patches in sync. This would be fine if it was just me, but that's not the case!
 - There's actually quite a lot of supporting code in copy-cat (for instance, handling `@LuaFunction` generation). Is that out of scope?

Other options:
 - Do nothing.
 - Update copy-cat's build process to provide the ROM as a separate bundle. tweaked.cc could then provide the "latest" ROM instead. This does mean the emulator may entirely break in some cases as the Java code lags behind - something to be careful of!
 - Refactor CC:T to avoid our reliance on patches. This should definitely be better post-https://github.com/cc-tweaked/CC-Tweaked/pull/1205 and related refactors, but some things are still painful.

   That said, refactoring some bits will be hard:
   - `ComputerThread` and `TimeoutState` are pretty tightly coupled, and `CobaltLuaMachine` has a heavy reliance on `TimeoutState` - not sure how to solve that.
   - `HTTPAPI` is _very_ reliant on Netty, and I don't think we can ever fix that. Maybe we're better off just replacing the whole HTTP API.

**Comment by SquidDev** [comment]
I've reduced the size of the patch set in b9fe03023c064e1d84e928c37398eaa2e7bb9320 and 8e610a95060e327261592205791a94c267f44f7b. I wonder if we should upstream a few additions (`String.trim`) before doing any further work.

**Comment by SquidDev** [comment]
f3b572b99acbdd82de63d30a1c78483d72cc3a74 and 341c3681a6bf625fca41b77fc939f069bd2f51a1 have reduced the number of patches further - we now only really need our patches for the HTTP API.

I think the next steps are to do all the following (no specific order):
 - Separate resources from classes (#36). I wonder if we could make this a little easier by splitting up `ArchiveMount` into one which doesn't use a cache. 
 - Upstream our chances to TeaVM.
 - Binary patch Cobalt to add-back too-long-without-yielding errors.

From there, we should then look at abstracting out the HTTP logic in core CC:T so we can use it without netty - this feels worth doing anyway, just to clean up the interface there a bit.

Once we've done that, all our source patches will be gone, so hopefully we can just binary patch the jar instead (it's just type remapping). I really don't want to doing source patches in the CC:T repo, as that means we need two versions of all the core classes, which is a nightmare to navigate.

**Comment by SquidDev** [comment]
True to form I did this in entirely the opposite order - 17c0fcfdebd66cad2d3c3d96da6f44efd4c26de0 now switches us to use binary patches.

> Upstream our changes to TeaVM.

I upstreamed the channel stuff earlier today. `Arrays.equals` should also be pretty easy, I want to confirm the behaviour of everything else before going any further.

I hit a weird issue today where the reference queue listener references an invalid method - think I need to do some further debugging there first, then report an issue.

> Binary patch Cobalt to add-back too-long-without-yielding errors.

This might be a little more awkward than I thought - I think we need to patch `CobaltLuaMachine`'s interrupt handler to call `TimeoutState.refresh()` as well.
